pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnectionPool.java

```diff
@@ -19,40 +19,18 @@
 package org.apache.pulsar.proxy.server;
 
 import io.netty.channel.EventLoopGroup;
-import java.io.IOException;
-import java.util.concurrent.ExecutionException;
+import io.netty.resolver.dns.DnsNameResolver;
+import java.util.Optional;
 import java.util.function.Supplier;
 import org.apache.pulsar.client.api.PulsarClientException;
 import org.apache.pulsar.client.impl.ClientCnx;
 import org.apache.pulsar.client.impl.ConnectionPool;
 import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ProxyConnectionPool extends ConnectionPool {
     public ProxyConnectionPool(ClientConfigurationData clientConfig, EventLoopGroup eventLoopGroup,
-            Supplier<ClientCnx> clientCnxSupplier) throws PulsarClientException {
-        super(clientConfig, eventLoopGroup, clientCnxSupplier);
+                               Supplier<ClientCnx> clientCnxSupplier, DnsNameResolver dnsNameResolver)
+            throws PulsarClientException {
+        super(clientConfig, eventLoopGroup, clientCnxSupplier, Optional.of(dnsNameResolver));
```

Comment:
Nit: is it worth removing this class and adding an additional constructor for the `ConnectionPool`?

Reply:
Yes, I did this.